src/Model/SnapUser.hs

```diff
@@ -0,0 +1,28 @@
+module Model.SnapUser where
```

Comment:
What about having the following code structure:
```
src/
    User/
        Types.hs
        Query.hs
        Persistence.hs
        Handlers.hs
```

Usually, that is the file structure for Haskell projects.

Inside the `Types.hs` will be your `User` type and its instances.